src/main/java/cloud/appuio/keycloak/extensions/mappers/ClaimToAttributeMapper.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @Override
+    public void updateBrokeredUser(KeycloakSession session, RealmModel realm, UserModel user, IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
+        this.extractClaimToAttribute(realm, mapperModel, user, new MapperConfig(mapperModel.getConfig()), context);
+    }
+
+    void extractClaimToAttribute(RealmModel realm, IdentityProviderMapperModel mapperModel, UserModel user, MapperConfig config, BrokeredIdentityContext context) {
+        // check if mapper is configured correctly.
+        if (config.getClaimName().equals("")) return;
+        if (config.getTargetAttributeKey().equals("")) return;
+
+        var claim = ClaimListExtractor.extractClaim(context, config.getClaimName());
+        if (claim.isEmpty()) {
+            logger.debugf(
+                    "Realm [%s], IdP [%s]: no [%s] claim for user [%s], ignoring...",
+                    realm.getName(),
+                    mapperModel.getIdentityProviderAlias(),
+                    config.getClaimName(),
+                    user.getUsername());
+            return;
+        }
+        assignClaimToAttribute(realm.getName(), mapperModel.getIdentityProviderAlias(), user, claim.get(), config);
+    }
+
+    void assignClaimToAttribute(String realmName, String identityProviderAlias, UserModel user, List<String> claimEntries, MapperConfig config) {
+        var attributes = user.getAttributeStream(config.getTargetAttributeKey()).collect(Collectors.toSet());
+        if (!config.enabledAttributeOverwrite()) {
+            var isAlreadyDefined = attributes.stream().anyMatch(value -> !"".equals(value));
+            if (isAlreadyDefined) {
+                logger.debugf("Realm [%s], IdP [%s]: Attribute [%s] is already set for user [%s]: [%s]",
+                        realmName,
+                        identityProviderAlias,
+                        config.getTargetAttributeKey(),
+                        user.getUsername(),
+                        String.join(", ", attributes)
+                );
+                return;
+            }
+        }
+        var formatter = new GroupNameFormatter()
+                .withToLowerCase(config.enabledToLowerCase())
+                .withTrimWhitespace(config.enabledTrimWhitespace())
+                .withTrimPrefix(config.getTrimPrefix());
+        var filteredGroups = claimEntries.stream()
+                .filter(this::ignoreEmptyEntries)
+                .filter(group -> ignoreEntriesThatMatchRegex(config, group))
+                .map(formatter::format)
+                .collect(Collectors.toList());
+
+        if (filteredGroups.size() != 1) {
+            logger.warnf("Realm [%s], IdP [%s]: Cannot reduce claim entries list to one entry for [%s]. Claim has following entries after reduction: [%s].",
+                    user.getUsername(),
+                    String.join(", ", filteredGroups));
+            return;
+        }
```

Comment:
Not sure about the log level here. Is it appropriate to log a warning? or should it be rather info or even debug?
The intention is to make it easier to figure out which users need manual actions. However, it's flawed:

* The logs can be lost after container restart
* It's getting logged every time the user logs in. This might be annoying, but counters the logs being lost a bit.